### DISCUSSION TITLE
OpenXR Integration

### DISCUSSION THREAD

**Discussion by AMZN-Liv** [o3de/o3de#6462] [Ideas]
Building applications for virtual and augmented reality hardware requires different input, rendering, and device-specific capabilities dependent on the characteristics of the specific target compared. Currently, building an application that targets a virtual or augmented reality device using O3DE would require significant engine modifications to be written. By integrating the OpenXR API into O3DE as an optional Gem for developers, we could create a route to enable users to target different a single API and take advantage of device-specific capabilities through a standard set of vendor extensions.

Khronos Group spec for OpenXR: https://www.khronos.org/openxr/ 

This is a general discussion to chat about interest and ideas related to a possible OpenXR integration. This initiative would likely span across SIGs (specifically SIG-CORE and SIG-GRAPHICS-AUDIO) with performance optimization work necessary for running at frame rates required to support HMD displays.

**Comment by jeremyong-az** [comment]
From a graphics perspective, things we need to consider:
- Exporting depth to the swapchain
- Relaxing depth precision from 24 to 16 bits for perf (as a scalability setting)
- Handling color spaces that aren't REC.709
- View instancing?VPRT

**Comment by HeadClot** [comment]
Hey - I really want to see OpenXR support in O3DE. Bit of a question from me however is how will input be handled? 
I am asking this about things like picking up an object for example.

**Comment by rodemen** [comment]
I am also very interested in OpenXR support in O3DE. The fact that we cannot use O3DE for VR games is preventing us from adopting the engine unfortunately.

**Comment by OBWANDO** [comment]
There are a few organizations that have OpenXR implementations with Vulkan support, but I think one key area is understanding what the needs would be to build the API interface between O3DE and OpenXR or at least identify where it would be plugged into.

**Comment by huashi59688** [comment]
Is there a roadmap to support OpenXR in O3DE?

**Reply by byrcolin** [reply]
The OpenXR for O3DE RFC is being worked on. Will be published shortly.

**Reply by HeadClot** [reply]
Hey @byrcolin - When the RFC goes up can you or someone else post it here?

**Reply by byrcolin** [reply]
Will do!

**Comment by moudgils** [comment]
RFC proposal uploaded here - https://github.com/o3de/sig-graphics-audio/issues/39.
